src/Compilers/CSharp/Test/Semantic/Semantics/NativeIntegerTests.cs

```diff
@@ -105,17 +103,23 @@ public abstract class ValueType
     public struct Void { }
     public struct Boolean { }
     public struct Int32 { }
-    public struct IntPtr
+    public interface IEquatable<T>
+    {
+        bool Equals(T other);
+    }
+    public struct IntPtr : IEquatable<IntPtr>
```

Comment:
>IEquatable<IntPtr> [](start = 27, length = 18)

Given that we have a custom code to match the type, I think we should cover negative cases as well, i.e.
- The name match, but containing namespace name doesn't
- Both names match, but System is not in a global namespace.
- IEquatable is a nested type and container's name is and is not System
- IEquatable is directly in a global namespace.
- IEquatable is not generic
- IEquatable has arity 2
- IEquatable is constructed with a "wrong" argument
- IEquatable is constructed with a nint for IntPtr and with nuint for UIntPtr
- IEquatable is constructed with a nuint for IntPtr and with nint for UIntPtr
- Etc.